test/dataframe.jl

```diff
@@ -325,25 +325,31 @@ module TestDataFrame
     df = DataFrame(a=Union{Int, Missing}[1, 2, 3], b=Union{Float64, Missing}[3.0, 4.0, 5.0])
     @test deleterows!(df, [2, 3]) === df
     @test df == DataFrame(a=[1], b=[3.0])
-
-    # describe
-    #suppress output and test that describe() does not throw
-    devmissing = Compat.Sys.isunix() ? "/dev/null" : "nul"
-    open(devmissing, "w") do f
-        @test nothing == describe(f, DataFrame(a=[1, 2], b=Any["3", missing]))
-        @test nothing ==
-              describe(f, DataFrame(a=Union{Int, Missing}[1, 2],
-                                    b=["3", missing]))
-        @test nothing ==
-              describe(f, DataFrame(a=CategoricalArray([1, 2]),
-                                    b=CategoricalArray(["3", missing])))
-        @test nothing == describe(f, [1, 2, 3])
-        @test nothing == describe(f, [1, 2, 3])
-        @test nothing == describe(f, CategoricalArray([1, 2, 3]))
-        @test nothing == describe(f, Any["1", "2", missing])
-        @test nothing == describe(f, ["1", "2", missing])
-        @test nothing == describe(f, CategoricalArray(["1", "2", missing]))
-    end
+    
+    # Test describe
+    struct MySuperLongNameForAStructIsThisToolong end
+    @testset "describe(df)" begin
+        describe_output = DataFrame(
+            variable = Vector{Any}([:number, :number_missing, :non_number, :non_number_missing, :long_struct]),
+            mean = [2.5, 2.0, nothing, nothing, nothing],
+            min = [1.0, 1.0, nothing, nothing, nothing],
+            median = [2.5, 2.0, nothing, nothing, nothing],
+            max = [4.0, 3.0, nothing, nothing, nothing],
+            nmissing = [nothing, 1, nothing, 1, nothing],
+            eltype= Vector{Any}([Int64, Int64, String, String, MySuperLongNameForAStructIsThisToolong])
+        )
+    
+        # Construct the test dataframe
+        df = DataFrame(number = [1, 2, 3, 4],
+            number_missing = [1,2, 3, missing],
```

Comment:
Wrong alignment.

It would also be nice to test the column types, and to have a test mixing e.g. numbers and Dates: the min and max will have heterogeneous types.

Reply:
This is the start of a larger conversation about edge-case types, I think. the `Date` type is not a subtype of number, so it acts like `Any`. 

Additionally, the behavior i currently have returns `nothing` for `nunique` on numeric types. On one hand, numeric variables are likely to have many values, making `nunique` an expensive operation. On the other hand, if you are using a small amount of integers, `nunique` makes sense. On the third hand, people might store `Point{Float64,Float64}` in a DataFrame, so `nunique` becomes even more expensive. 

Not to even mention factor variables. A numeric vector that is then coded as a categorical variable ceases to be a numeric. This might be annoying if you want to explore survey results coded 1-5, but then use them as categorical data in `GLM`. I'm sure there are tons more types that are in between `Numeric` and not, and I'm not entirely sure how to approach all of those cases.

Reply:
Of course, I forgot this was limited to numbers. Yet there can be some "weird" number types, so it's not a bad idea to use a comprehension so that the element type is chosen automatically.

Good point about the fact that the number of unique values could become too high even for non-`Number` types (`Date` being a case in point). At the extreme, the computer could run out of memory, which would be really bad. I guess the best we can do would be to use a custom `nunique` function which would replicate `unique`, but stop after a large number of unique values has been found.

Regarding numeric columns with a small number of unique values, I'm not sure there's much we can do. People should probably use `CategoricalArray` if they want these variables to be treated as categorical.

As you suggest, it could make sense to use `try... catch` (and/or `applicable`). That's not hard, just verbose. Probably better leave that for another PR anyway.

Reply:
Submitting this PR now, but will open up another one soon with `try..catch`. Should be very simple since it's just `get_stats` that's being changed.